src/compiler/msvc.rs

```diff
@@ -430,6 +432,19 @@ fn compile<T>(creator: &T,
                 (cacheable, output)
             }))
         }
+    });
+
+    // If the `-showIncludes` command line option was originally passed we need
+    // to be sure to ship the output from the preprocessor as the actual
+    // result of this compilation.
+    let mut extra_stderr = Vec::new();
+    if parsed_args.msvc_show_includes {
+        extra_stderr = preprocessor_result.stderr;
+    }
+    Box::new(ret.map(|(cacheable, mut output)| {
+        let prev = mem::replace(&mut output.stderr, extra_stderr);
+        output.stderr.extend(prev);
```

Comment:
I don't think this is actually correct, and this is probably what breaks the configure test in the Firefox build. See my comment in `detect_showincludes_prefix`:
https://github.com/mozilla/sccache/blob/7ae4c259873d6bcbb60ad58bd9735581f7ddbe7a/src/compiler/msvc.rs#L104

I just did a simple local test with MSVC2015 and the `-showIncludes` output goes to stdout when compiling:
```
ted@DESKTOP-32V9ND8 /c/build
$ cl -Fonul -nologo -showIncludes -c test.c > /tmp/cl.stdout 2>/tmp/cl.stderr

ted@DESKTOP-32V9ND8 /c/build
$ cat /tmp/cl.stdout
test.c
Note: including file: C:\Program Files (x86)\Windows Kits\10\include\10.0.10586.0\ucrt\stdio.h
Note: including file:  C:\Program Files (x86)\Windows Kits\10\include\10.0.10586.0\ucrt\corecrt.h
Note: including file:   C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\INCLUDE\vcruntime.h
Note: including file:    C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\INCLUDE\sal.h
Note: including file:     C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\INCLUDE\ConcurrencySal.h
Note: including file:    C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\INCLUDE\vadefs.h
Note: including file:  C:\Program Files (x86)\Windows Kits\10\include\10.0.10586.0\ucrt\corecrt_wstdio.h
Note: including file:   C:\Program Files (x86)\Windows Kits\10\include\10.0.10586.0\ucrt\corecrt_stdio_config.h
```

When running the preprocessor it definitely goes to stderr.

Unfortunately I think this means we'll have to do something slightly more complex here, like parse the preprocessor stderr to get just the showIncludes lines, and then put just those in the compiler stdout.